include/classes/AMZNBS/Admin.class.php

```diff
@@ -247,16 +249,17 @@ public function update_options($form) {
         // printf("<pre>In update_options()\OPTS: %s\naction = %s</pre>",print_r($opts,1),$_REQUEST['action']);
         $this->options['affiliate_id'] = $opts['affiliate_id'];
         $this->options['country_id'] = $opts['country_id'];
-        $this->options['default'] = SGW_BESTSELLERS;
-        // update the default asins, if present
-        if ($test = $this->normalize_asin_list($opts['default'])) {
-          $this->options['default'] = $test;
-        }
-        update_option(SGW_PLUGIN_OPTTIONS,$this->options);
+
+        $this->update_default_asins($opts['default']);
+        // $this->update_new_asins($opts['new']);
+        // $this->update_existing_asins($opts['posts']);
+
         // update the newly added ASINs
         if ($opts['new']) {
           foreach ($opts['new'] as $id=>$hash) {
             if ($test = $this->normalize_asin_list($hash['asin'])) {
+              // TODO: Hook into this to pre-fetch Images for the listed ASINs, but we want to store in meta as a hash that can be unpacked
```

Comment:
Remove this todo, as this was implemented in a different way.  For performance reasons, asins will not be fetched when being added, but lazily when the post to which they are associated is loaded the first time.